### ISSUE TITLE
env interpolation sometimes randomly fails

### ISSUE THREAD

**Issue by sstarcher** [roboll/helmfile#37]
I have a loop that runs helmfile diff 100 times.  Sometimes the following value will output true, but the majority of the time it has the correct value
```
      - name: config.s3.bucket
        value: '{{ env "S3_BUCKET" }}'
```

* exec: helm diff chart chart config.s3.bucket=true 
* exec: helm diff chart chart config.s3.bucket=bucket-name

**Comment by mumoshu** [comment]
@sstarcher Interesting! Out of curiosity, I ran `go test -race` against a test case like:

```golang
func TestRenderTemplateString(t *testing.T) {
	for i := 0; i<100; i++ {
		for j := 0; j<100; j++ {
			name := fmt.Sprintf("HELMFILE_TEST_ENV_%d", j)
			value := fmt.Sprintf("test_env_%d", j)
			os.Setenv(name, value)
			{
				test1, err := renderTemplateString(fmt.Sprintf("{{ env \"%s\" }}", name))
				if err != nil {
					t.Errorf("unexpected error: %v", err)
				}
				if test1 != value {
					t.Errorf("unexpected value read from env var: %v", test1)
				}
			}
		}
	}
}
```

Which (of course?) doesn't fail.

Also, looking into the renderTemplate code:

```
for _, value := range chart.Values {
		valfile := filepath.Join(basePath, value)
		valfileRendered, err := renderTemplateString(valfile)
		if err != nil {
			return nil, err
		}
		flags = append(flags, "--values", valfileRendered)
	}
	if len(chart.SetValues) > 0 {
		val := []string{}
		for _, set := range chart.SetValues {
			renderedValue, err := renderTemplateString(set.Value)
			if err != nil {
				return nil, err
			}
			val = append(val, fmt.Sprintf("%s=%s", set.Name, renderedValue))
		}
		flags = append(flags, "--set", strings.Join(val, ","))
	}
```

I see nothing suspicious in regard to race. Funny.

**Comment by sstarcher** [comment]
ya, currently trying to reproduce this one in a simpler means.

**Comment by sstarcher** [comment]
got it to happen again, but only when using multiple charts.  This time it happend on a different variable.  It's likely related to the templating race condition.

**Comment by mumoshu** [comment]
Ah, thanks! Helmfile does run templating concurrently in HelmState.SyncCharts. The fix in #38 would work then.

**Comment by sstarcher** [comment]
Looks like #38 also fixes this issue
